src/Service/Startup.cs

```diff
@@ -213,50 +222,61 @@ public void ConfigureServices(IServiceCollection services)
         /// when determining whether to allow introspection requests to proceed.
         /// </summary>
         /// <param name="services">Service Collection</param>
-        private void AddGraphQLService(IServiceCollection services)
+        private void AddGraphQLService(IServiceCollection services, int? depthLimit = null)
         {
-            services.AddGraphQLServer()
-                    .AddHttpRequestInterceptor<DefaultHttpRequestInterceptor>()
-                    .ConfigureSchema((serviceProvider, schemaBuilder) =>
-                    {
-                        GraphQLSchemaCreator graphQLService = serviceProvider.GetRequiredService<GraphQLSchemaCreator>();
-                        graphQLService.InitializeSchemaAndResolvers(schemaBuilder);
-                    })
-                    .AddHttpRequestInterceptor<IntrospectionInterceptor>()
-                    .AddAuthorization()
-                    .AllowIntrospection(false)
-                    .AddAuthorizationHandler<GraphQLAuthorizationHandler>()
-                    .AddErrorFilter(error =>
+            IRequestExecutorBuilder server = services.AddGraphQLServer()
+                .AddHttpRequestInterceptor<DefaultHttpRequestInterceptor>()
+                .ConfigureSchema((serviceProvider, schemaBuilder) =>
+                {
+                    GraphQLSchemaCreator graphQLService = serviceProvider.GetRequiredService<GraphQLSchemaCreator>();
+                    graphQLService.InitializeSchemaAndResolvers(schemaBuilder);
+                })
+                .AddHttpRequestInterceptor<IntrospectionInterceptor>()
+                .AddAuthorization()
+                .AllowIntrospection(false)
+                .AddAuthorizationHandler<GraphQLAuthorizationHandler>();
+
+            // Conditionally adds a maximum depth rule to the GraphQL queries/mutation.
```

Comment:
> // Conditionally adds a maximum depth rule to the GraphQL queries/mutation.

 Need to be more specific: Max Query Depth on a mutation's selection set, not mutation input?